app/controllers/dams_resource_controller.rb

```diff
@@ -188,7 +190,22 @@ def solr_escape (str)
     pattern = /(\+|\-|\&\&|\|\||\!|\(\)|\{\}|\[|\]|\^|\"|\~|\*|\?|\:|\\)/
     str.gsub(pattern){|match|"\\"  + match} 
   end
-
+  
+  def replace_config_file (file_name)
+    copy_config_file (file_name)
+    config = ERB.new(IO.read("#{APP_ROOT}/config/#{file_name}")).result(binding)
+    File.open("#{APP_ROOT}/config/#{file_name}", "w") do |f|
+      f.write(config)
+    end
+  end
+  
+  def copy_config_file (file_name)
+    data = File.read("#{APP_ROOT}/config/#{Rails.configuration.share_notify_sample}")
+    File.open("#{APP_ROOT}/config/#{file_name}", "w") do |fi|
+      fi.write(data)
+    end
```

Comment:
@VivianChu Is this repeated block of code as it is in #replace_config_file? I'm not sure why do you need it?